src/main/java/vazkii/botania/api/subtile/TileEntityBindableSpecialFlower.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * This class is distributed as part of the Botania Mod.
+ * Get the Source Code in github:
+ * https://github.com/Vazkii/Botania
+ *
+ * Botania is Open Source and distributed under the
+ * Botania License: http://botaniamod.net/license.php
+ */
+package vazkii.botania.api.subtile;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.NbtUtils;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+
+import vazkii.botania.api.block.IWandBindable;
+import vazkii.botania.common.core.helper.MathHelper;
+
+import javax.annotation.Nullable;
+
+import java.util.Objects;
+
+/**
+ * Superclass of all flowers which are bound to something with the Wand of the Forest,
+ * such as generating flowers to mana collectors, or mana-using functional flowers to pools.
+ * Implements bindability logic common to both types of flower.
+ */
+public abstract class TileEntityBindableSpecialFlower<T> extends TileEntitySpecialFlower implements IWandBindable {
+	protected @Nullable BlockPos bindingPos = null;
+	private final Class<T> tileClass;
+
+	private static final String TAG_BINDING = "binding";
+
+	public TileEntityBindableSpecialFlower(BlockEntityType<?> type, BlockPos pos, BlockState state, Class<T> tileClass) {
```

Comment:
tileClass -> bindClass and maybe document that it's the class that we want  to bind to

Reply:
how's this?